exporters/otlp/otlpmetric/internal/otlpconfig/options.go

```diff
@@ -72,24 +74,52 @@ type (
 	}
 )
 
-func NewDefaultConfig() Config {
-	c := Config{
+// NewHTTPConfig returns a new Config with all settings applied from opts and
+// any unset setting using the default HTTP config values.
+func NewHTTPConfig(opts ...HTTPOption) Config {
+	cfg := Config{
 		Metrics: SignalConfig{
-			Endpoint:    fmt.Sprintf("%s:%d", DefaultCollectorHost, DefaultCollectorPort),
+			Endpoint:    fmt.Sprintf("%s:%d", DefaultCollectorHost, DefaultCollectorHTTPPort),
 			URLPath:     DefaultMetricsPath,
 			Compression: NoCompression,
 			Timeout:     DefaultTimeout,
 		},
 		RetryConfig: retry.DefaultConfig,
 	}
+	cfg = ApplyHTTPEnvConfigs(cfg)
+	for _, opt := range opts {
+		cfg = opt.ApplyHTTPOption(cfg)
+	}
 
-	return c
+	for pathPtr, defaultPath := range map[*string]string{
+		&cfg.Metrics.URLPath: DefaultMetricsPath,
+	} {
+		tmp := strings.TrimSpace(*pathPtr)
+		if tmp == "" {
+			tmp = defaultPath
+		} else {
+			tmp = path.Clean(tmp)
+			if !path.IsAbs(tmp) {
+				tmp = fmt.Sprintf("/%s", tmp)
+			}
+		}
+		*pathPtr = tmp
```

Comment:
I am not sure why to use this strange `range` style to format `cfg.Metrics.URLPath`.

Codes like this can do the same job but clear.

```go
	tmp := strings.TrimSpace(cfg.Metrics.URLPath)
	if tmp == "" {
		tmp = DefaultMetricsPath
	} else {
		tmp = path.Clean(tmp)
		if !path.IsAbs(tmp) {
			tmp = fmt.Sprintf("/%s", tmp)
		}
	}
	cfg.Metrics.URLPath = tmp
```

Reply:
Sure. I was just copying this from the unified location of the `NewClient` code to make it clear the main purpose of this PR, but I can update these as well.